### PR TITLE
[Improvement] Support for openJDk Nashorn, JDK 11 compilation and Mockito 2

### PR DESCRIPTION
**Pull request by Thumimku**
### Proposed changes in this pull request

Issue:
https://github.com/wso2/product-is/issues/13463

We have improved the Identity Framework for the following artifacts.

- Since Jashorn is deprecated, we are going to use [openjdk/Nashorn](https://github.com/openjdk/nashorn).
- To support OpenJDK nashorn we are going to add compilation support for JDK 11.
- We are bumping from mockito 1 to mockito 2 to support JDK 11 compilation.

### When should this PR be merged

**After this PR merged to main branch, we can't Build the branch on JDK 8**

### Follow up actions

- Run PR builder for this branch on JDK 11 to test the integration test of Product IS. 

### Steps to follow

1. Download and set up Open JDK 11.
2. Fetch and create a branch for this PR. ([Hint](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#modifying-an-inactive-pull-request-locally))
3. Try to build in JDK 11.

Build Final Output
```
[INFO] Reactor Summary for WSO2 Carbon - Platform Aggregator Pom 5.20.410-SNAPSHOT:
[INFO] 
[INFO] WSO2 Carbon - Platform Aggregator Pom .............. SUCCESS [  4.630 s]
[INFO] WSO2 Carbon - Identity Core Aggregator Module ...... SUCCESS [  0.986 s]
[INFO] WSO2 Carbon - Identity Base ........................ SUCCESS [ 10.200 s]
[INFO] WSO2 Carbon - Identity Core ........................ SUCCESS [ 19.315 s]
[INFO] WSO2 Carbon - Identity Common Test Util ............ SUCCESS [  6.545 s]
[INFO] WSO2 Carbon - Service Stubs - Aggregator Module .... SUCCESS [  0.064 s]
[INFO] WSO2 Carbon - Security Stub ........................ SUCCESS [  9.373 s]
[INFO] WSO2 Carbon - Claim Management Stub ................ SUCCESS [  5.392 s]
[INFO] WSO2 Carbon - Identity Claim Metadata Management Stub SUCCESS [  6.071 s]
[INFO] WSO2 Carbon - Identity Application Management Stub . SUCCESS [ 11.342 s]
[INFO] WSO2 Carbon - Identity Application Authentication Framework Stub SUCCESS [  3.939 s]
[INFO] WSO2 Carbon - Entitlement Stub ..................... SUCCESS [ 15.101 s]
[INFO] WSO2 Carbon - Identity Management Stub ............. SUCCESS [ 16.847 s]
[INFO] WSO2 Carbon - User Profile Management Stub ......... SUCCESS [  7.524 s]
[INFO] WSO2 Carbon - Identity User Registration Stub ...... SUCCESS [  5.927 s]
[INFO] WSO2 Carbon - User Store Config Stub ............... SUCCESS [  6.668 s]
[INFO] WSO2 Carbon - Identity Provider Mgt Stub ........... SUCCESS [  8.075 s]
[INFO] WSO2 Carbon - Service Principle Management Stub .... SUCCESS [  5.333 s]
[INFO] WSO2 Carbon - User Management Stub ................. SUCCESS [ 12.435 s]
[INFO] WSO2 Carbon - User Mgt Workflow Stub ............... SUCCESS [  4.945 s]
[INFO] WSO2 Carbon - Workflow Mgt Stub .................... SUCCESS [  9.199 s]
[INFO] WSO2 Carbon - User Store Count Stub ................ SUCCESS [  6.038 s]
[INFO] WSO2 Carbon - Identity Governance Stub ............. SUCCESS [  5.354 s]
[INFO] WSO2 Carbon - Functions Library Management Stub .... SUCCESS [  6.451 s]
[INFO] WSO2 Carbon - Identity Application Default Authentication Sequence Management Stub SUCCESS [  9.918 s]
[INFO] WSO2 Carbon - Authentication Framework Aggregator Module SUCCESS [  1.333 s]
[INFO] WSO2 Carbon - Identity Application Authentication Endpoint(Utils) SUCCESS [  8.962 s]
[INFO] WSO2 Carbon - Application Management Aggregator Module SUCCESS [  1.114 s]
[INFO] WSO2 Carbon - Identity Application Common Bundle ... SUCCESS [ 20.117 s]
[INFO] WSo2 Carbon - User Management Aggregator Module .... SUCCESS [  1.212 s]
[INFO] WSO2 Carbon - User Management Common ............... SUCCESS [  3.530 s]
[INFO] WSO2 Carbon - User Management ...................... SUCCESS [ 12.690 s]
[INFO] WSO2 Carbon - Identity Event Module ................ SUCCESS [  1.286 s]
[INFO] WSO2 Carbon - Identity Event ....................... SUCCESS [  8.577 s]
[INFO] WSO2 Carbon - Role Manager ......................... SUCCESS [  1.171 s]
[INFO] WSO2 Carbon - Role management ...................... SUCCESS [ 24.968 s]
[INFO] WSo2 Carbon - User Store Aggregator Module ......... SUCCESS [  1.223 s]
[INFO] WSO2 Carbon - User store Configuration Component ... SUCCESS [ 11.575 s]
[INFO] WSO2 Carbon - Claim Management Aggregator Module ... SUCCESS [  1.130 s]
[INFO] WSO2 Carbon - Identity Claim Metadata Management BE  SUCCESS [ 16.928 s]
[INFO] WSO2 Carbon - Identity Provider Management Aggregator Module SUCCESS [  1.640 s]
[INFO] WSO2 Carbon - Identity Provider Management Bundle .. SUCCESS [ 28.969 s]
[INFO] WSO2 Carbon - Security Aggregator Module ........... SUCCESS [  1.314 s]
[INFO] WSO2 Carbon - Security BE .......................... SUCCESS [ 13.979 s]
[INFO] central-logger ..................................... SUCCESS [  1.418 s]
[INFO] org.wso2.carbon.identity.central.log.mgt ........... SUCCESS [  4.017 s]
[INFO] WSO2 Identity- Application Management Component .... SUCCESS [ 40.046 s]
[INFO] WSO2 Carbon - User Profile Management .............. SUCCESS [  4.606 s]
[INFO] WSO2 Carbon - Functions Library Manager ............ SUCCESS [  1.561 s]
[INFO] WSO2 Carbon - Functions Library Manager Server Component SUCCESS [ 14.432 s]
[INFO] WSO2 Carbon - Multi Attribute Aggregator Module .... SUCCESS [  1.463 s]
[INFO] WSO2 Carbon - Multi Attribute Login Management ..... SUCCESS [  3.307 s]
[INFO] WSO2 Carbon - Claim Management BE .................. SUCCESS [  3.627 s]
[INFO] WSO2 Carbon - Identity Application Authentication Framework SUCCESS [ 59.234 s]
[INFO] WSO2 Carbon - Identity Authenticators .............. SUCCESS [  1.423 s]
[INFO] WSO2 Carbon - Thrift Authenticator ................. SUCCESS [  1.010 s]
[INFO] WSO2 Carbon - Thrift based Authentication Service .. SUCCESS [  4.474 s]
[INFO] WSO2 Carbon - Entitlement Aggregator Module ........ SUCCESS [  0.924 s]
[INFO] WSO2 Carbon - Identity Entitlement Common .......... SUCCESS [  4.362 s]
[INFO] WSO2 Carbon - Notification Management Aggregator Module SUCCESS [  0.962 s]
[INFO] WSO2 Carbon - Notification Management BE ........... SUCCESS [  3.102 s]
[INFO] WSO2 Carbon - Identity Entitlement ................. SUCCESS [  9.959 s]
[INFO] WSO2 Carbon - Identity Entitlement UI .............. SUCCESS [  5.970 s]
[INFO] WSO2 Carbon - Entitlement REST API ................. SUCCESS [  8.798 s]
[INFO] WSO2 Carbon - Entitlement REST ..................... SUCCESS [  1.669 s]
[INFO] WSO2 Carbon - identity tree menu structure ......... SUCCESS [  0.208 s]
[INFO] WSO2 Carbon - Application Management UI ............ SUCCESS [ 11.939 s]
[INFO] WSO2 Carbon - User Management UI ................... SUCCESS [  4.931 s]
[INFO] WSO2 Carbon - User Profile Management UI ........... SUCCESS [  3.097 s]
[INFO] WSO2 Carbon - Identity User Registration ........... SUCCESS [  2.829 s]
[INFO] WSO2 Carbon - Captcha Management Module ............ SUCCESS [  1.210 s]
[INFO] WSO2 Carbon - Captcha Managment .................... SUCCESS [  2.096 s]
[INFO] WSO2 Carbon - Identity Management Aggregator Module  SUCCESS [  0.705 s]
[INFO] WSO2 Carbon - Identity Management .................. SUCCESS [ 10.240 s]
[INFO] WSO2 Carbon - Identity Management UI ............... SUCCESS [  2.674 s]
[INFO] WSO2 Carbon - Identity Management Endpoint Util .... SUCCESS [ 12.355 s]
[INFO] WSO2 Carbon - Provisioning Aggregator Module ....... SUCCESS [  1.250 s]
[INFO] WSO2 Carbon - Identity Provisioning ................ SUCCESS [  7.653 s]
[INFO] WSO2 Carbon - Identity User Store Configuration Deployment SUCCESS [  2.689 s]
[INFO] WSO2 Carbon - Identity User Store Configuration UI . SUCCESS [  3.005 s]
[INFO] WSO2 Carbon - Identity User Store Counts ........... SUCCESS [  4.894 s]
[INFO] WSO2 Carbon - Identity Provider Management UI Bundle SUCCESS [ 10.359 s]
[INFO] WSO2 Carbon - Identity Claim Metadata Management UI  SUCCESS [  3.083 s]
[INFO] WSO2 Carbon - Claim Management UI .................. SUCCESS [  2.480 s]
[INFO] WSO2 Carbon - Directory server manager. Responsible for defining and managing servers in LDAP. SUCCESS [  1.687 s]
[INFO] WSO2 Carbon - Server Management Common ............. SUCCESS [  1.443 s]
[INFO] WSO2 Carbon - Directory server manager ............. SUCCESS [  3.562 s]
[INFO] WSO2 Carbon - Service Principle Management UI ...... SUCCESS [  2.461 s]
[INFO] WSO2 Carbon - Security UI .......................... SUCCESS [  8.973 s]
[INFO] WSO2 Identity Workflow Management Parent Module .... SUCCESS [  1.399 s]
[INFO] WSO2 Carbon - Workflow Mgt ......................... SUCCESS [  6.273 s]
[INFO] WSO2 Carbon - Workflow Management UI ............... SUCCESS [  3.414 s]
[INFO] WSO2 Carbon - Policy Editor Aggregator Module ...... SUCCESS [  1.283 s]
[INFO] WSO2 Carbon - Policy Editor BE ..................... SUCCESS [  2.563 s]
[INFO] WSO2 Carbon - Policy Editor UI ..................... SUCCESS [  0.585 s]
[INFO] WSO2 Carbon - Identity Consent Mgt Module .......... SUCCESS [  0.747 s]
[INFO] WSO2 Carbon Identity- Consent Mgt BE ............... SUCCESS [  6.854 s]
[INFO] WSO2 Carbon - Functions Library Manager UI Component SUCCESS [  2.285 s]
[INFO] WSO2 Carbon - Configuration Management Aggregator Module SUCCESS [  1.268 s]
[INFO] WSO2 Carbon - Configuration Management Core ........ SUCCESS [ 10.418 s]
[INFO] WSO2 Carbon - Template Manager ..................... SUCCESS [  1.263 s]
[INFO] WSO2 Carbon - Identity Template Management ......... SUCCESS [ 11.367 s]
[INFO] WSO2 Carbon - Template Management UI ............... SUCCESS [  3.175 s]
[INFO] WSO2 Carbon - Configuration Management API ......... SUCCESS [  4.321 s]
[INFO] WSO2 Carbon - Configuration Management Endpoint .... SUCCESS [  1.754 s]
[INFO] WSO2 Carbon - Per User Functionality Management Aggregator Module SUCCESS [  0.670 s]
[INFO] WSO2 Carbon - Per User Functionality Management .... SUCCESS [ 14.004 s]
[INFO] WSO2 Carbon - CORS Management Aggregator Module .... SUCCESS [  1.174 s]
[INFO] WSO2 Carbon - CORS Management Core ................. SUCCESS [ 11.073 s]
[INFO] unique-claim-user-operation-event-listener ......... SUCCESS [  4.816 s]
[INFO] WSO2 Carbon - Secret Management Aggregator Module .. SUCCESS [  0.911 s]
[INFO] WSO2 Carbon - Secret Management Core ............... SUCCESS [ 10.583 s]
[INFO] WSO2 Carbon - Security Management Feature Aggregator Module SUCCESS [  1.604 s]
[INFO] WSO2 Carbon - Security Management Core Feature ..... SUCCESS [  1.808 s]
[INFO] WSO2 Carbon - Security Management UI Feature ....... SUCCESS [  1.790 s]
[INFO] Key Store Management Feature ....................... SUCCESS [  1.722 s]
[INFO] WSO2 Carbon - Claim Management Feature Aggregator Module SUCCESS [  0.805 s]
[INFO] Claim Management UI Feature ........................ SUCCESS [  1.544 s]
[INFO] Claim Management Core Feature ...................... SUCCESS [  1.466 s]
[INFO] Claim Management Feature ........................... SUCCESS [  1.525 s]
[INFO] WSO2 Carbon - Identity Core Feature Aggregator Module SUCCESS [  0.873 s]
[INFO] Identity Core UI Feature ........................... SUCCESS [  1.033 s]
[INFO] Identity Core Server Feature ....................... SUCCESS [  1.205 s]
[INFO] Identity Core Feature .............................. SUCCESS [  1.327 s]
[INFO] WSO2 Carbon - Identity Mgt Feature Aggregator Module SUCCESS [  0.813 s]
[INFO] Identity Mgt Server Feature ........................ SUCCESS [  1.168 s]
[INFO] Identity Event Feature ............................. SUCCESS [  1.360 s]
[INFO] WSO2 Carbon - Identity Mgt Feature Aggregator Module SUCCESS [  0.790 s]
[INFO] Identity Mgt UI Feature ............................ SUCCESS [  1.179 s]
[INFO] Identity Mgt Server Feature ........................ SUCCESS [  1.704 s]
[INFO] Account Recovery and Credential Management Feature . SUCCESS [  1.845 s]
[INFO] WSO2 Carbon - XACML Feature Aggregator Module ...... SUCCESS [  0.912 s]
[INFO] Identity XACML Server Feature ...................... SUCCESS [  2.049 s]
[INFO] Identity XACML UI Feature .......................... SUCCESS [  1.232 s]
[INFO] XACML Feature ...................................... SUCCESS [  1.707 s]
[INFO] WSO2 Carbon - Application Autentication Framework Feature Aggregator Module SUCCESS [  0.736 s]
[INFO] Application Authentication Framework Server Feature  SUCCESS [  2.039 s]
[INFO] WSO2 Carbon - Application Management Feature Aggregator Module SUCCESS [  0.917 s]
[INFO] Identity Application Management Server Feature ..... SUCCESS [  1.382 s]
[INFO] Identity Application Management UI Feature ......... SUCCESS [  1.321 s]
[INFO] Identity Application Managment Feature ............. SUCCESS [  1.418 s]
[INFO] WSO2 Carbon - Identity Provisioning Feature ........ SUCCESS [  0.820 s]
[INFO] Identity Provisioning Server Feature ............... SUCCESS [  1.948 s]
[INFO] WSO2 Carbon - Identity Notification Mgt Module ..... SUCCESS [  0.820 s]
[INFO] Notification-mgt server feature .................... SUCCESS [  0.982 s]
[INFO] Notification Mgt Feature ........................... SUCCESS [  1.048 s]
[INFO] WSO2 Carbon - Identity Provider Management Feature Aggregator Module SUCCESS [  0.720 s]
[INFO] Identity Provider Management UI Feature ............ SUCCESS [  1.723 s]
[INFO] Identity Provider Management Server Feature ........ SUCCESS [  1.292 s]
[INFO] Trusted Identity Provider Management Feature ....... SUCCESS [  1.598 s]
[INFO] WSO2 Carbon - LDAP Server Principle Management ..... SUCCESS [  0.935 s]
[INFO] WSO2 Carbon - Service Principle Management ......... SUCCESS [  1.440 s]
[INFO] WSO2 Carbon - Service Principle Management UI ...... SUCCESS [  1.280 s]
[INFO] WSO2 Carbon - Service Principle Management Feature . SUCCESS [  1.589 s]
[INFO] WSO2 Carbon - User Manager Aggregator Module ....... SUCCESS [  0.803 s]
[INFO] User Manager UI Feature ............................ SUCCESS [  1.537 s]
[INFO] User Manager Server Feature ........................ SUCCESS [  1.182 s]
[INFO] User Management Feature ............................ SUCCESS [  1.229 s]
[INFO] Identity User Profiles UI Feature .................. SUCCESS [  0.982 s]
[INFO] Identity User Profiles Server Feature .............. SUCCESS [  1.219 s]
[INFO] User Profiles Feature .............................. SUCCESS [  1.333 s]
[INFO] Identity user Registration Server Feature .......... SUCCESS [  1.129 s]
[INFO] Identity User Registration UI Feature .............. SUCCESS [  0.974 s]
[INFO] Identity User Registration Feature ................. SUCCESS [  1.280 s]
[INFO] WSO2 Carbon - Role Management Feature Aggregator Module SUCCESS [  0.785 s]
[INFO] WSO2 Carbon - Role Management Feature .............. SUCCESS [  1.011 s]
[INFO] WSO2 Carbon - Workflow Management Feature Aggregator Module SUCCESS [  0.969 s]
[INFO] WSO2 Carbon - Workflow Management Core Feature ..... SUCCESS [  1.113 s]
[INFO] WSO2 Carbon - Workflow Management UI Feature ....... SUCCESS [  1.207 s]
[INFO] Workflow Management Feature ........................ SUCCESS [  1.394 s]
[INFO] WSO2 Carbon Identity - Authorization Feature Category SUCCESS [  0.859 s]
[INFO] WSO2 Carbon Identity - Inbound Authentication Feature Category SUCCESS [  0.805 s]
[INFO] WSO2 Carbon Identity - Inbound Provisioning Feature Category SUCCESS [  0.984 s]
[INFO] WSO2 Carbon Identity - Keystore Management Feature Category SUCCESS [  1.035 s]
[INFO] WSO2 Carbon Identity - Notification Management Feature Category SUCCESS [  0.931 s]
[INFO] WSO2 Carbon Identity - Outbound Authentication Feature Category SUCCESS [  0.993 s]
[INFO] WSO2 Carbon Identity - Outbound Provisioning Feature Category SUCCESS [  0.932 s]
[INFO] WSO2 Carbon Identity - User Management Feature Category SUCCESS [  0.822 s]
[INFO] WSO2 Carbon Identity - Workflow Feature Category ... SUCCESS [  0.909 s]
[INFO] WSO2 Carbon Identity - Feature Categories Module ... SUCCESS [  0.901 s]
[INFO] WSO2 Carbon - Authenticator Feature Aggregator Module SUCCESS [  0.762 s]
[INFO] WSO2 Carbon - Thrift Authentication Aggregate Feature SUCCESS [  0.878 s]
[INFO] Identity Thrift Authentication Feature ............. SUCCESS [  1.503 s]
[INFO] WSO2 Carbon - Identity Consent Aggregator Module ... SUCCESS [  0.978 s]
[INFO] Identity Consent Management Server Feature ......... SUCCESS [  1.509 s]
[INFO] WSO2 Carbon - Function Library Management Feature Aggregator Module SUCCESS [  0.822 s]
[INFO] Function Library Management Server Feature ......... SUCCESS [  0.899 s]
[INFO] Function Library Management UI Feature ............. SUCCESS [  1.033 s]
[INFO] Function Library Managment Feature ................. SUCCESS [  1.106 s]
[INFO] WSO2 Carbon - Identity Template Management Feature Aggregator Module SUCCESS [  0.785 s]
[INFO] Template Management Server Feature ................. SUCCESS [  1.280 s]
[INFO] Template Mgt UI Feature ............................ SUCCESS [  1.378 s]
[INFO] Template Management Feature ........................ SUCCESS [  1.434 s]
[INFO] WSO2 Carbon - Configuration Management Feature Aggregator Module SUCCESS [  0.984 s]
[INFO] WSO2 Carbon - Configuration Management Feature ..... SUCCESS [  1.746 s]
[INFO] WSO2 Carbon - Identity User Functionality Management Feature Aggregator Module SUCCESS [  0.904 s]
[INFO] User Functionality Management Server Feature ....... SUCCESS [  1.194 s]
[INFO] User Functionality Management Feature .............. SUCCESS [  1.322 s]
[INFO] WSO2 Carbon - CORS Management Feature Aggregator Module SUCCESS [  0.726 s]
[INFO] WSO2 Carbon - CORS Management Feature .............. SUCCESS [  1.377 s]
[INFO] WSO2 Carbon - Multi Attribute Login Module ......... SUCCESS [  1.114 s]
[INFO] Unique Claim Mgt Feature ........................... SUCCESS [  0.903 s]
[INFO] Unique Claim Mgt Feature ........................... SUCCESS [  1.730 s]
[INFO] WSO2 Carbon - User Store Aggregator Module ......... SUCCESS [  0.972 s]
[INFO] Userstore Configuration Feature .................... SUCCESS [  0.741 s]
[INFO] WSO2 Carbon - Secret Management Feature Aggregator Module SUCCESS [  0.906 s]
[INFO] WSO2 Carbon - Secret Management Feature ............ SUCCESS [  1.041 s]
[INFO] WSO2 Carbon - Central Logger Feature Aggregator Module SUCCESS [  0.720 s]
[INFO] WSO2 Carbon - Central Log Management Feature ....... SUCCESS [  1.017 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14:20 min
[INFO] Finished at: 2022-05-31T18:10:01+05:30
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/pom.xml [hunk 1]
```diff
@@ -120,29 +120,33 @@
...
-        </dependency>
         <dependency>
             <groupId>org.jacoco</groupId>
             <artifactId>org.jacoco.agent</artifactId>
             <classifier>runtime</classifier>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>${mockito.version}</version>
```
Comment (Thumimku) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/pom.xml:131]: No need to put version explicitly

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtilTest.java [hunk 2]
```diff
@@ -353,6 +359,7 @@ public Object[][] updatePermissionDataProvider() {
        return new Object[][]{
                {new String[]{PERMISSION, ""}, 1},
                {new String[]{}, 0}
+
```
Comment (Thumimku) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtilTest.java:362]: New line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 3]
```diff
@@ -266,8 +266,8 @@ public void testGetStepHandlerExistHandler() {
     public void testGetStepHandlerNonExistHandler() {
 
         ConfigurationFacade.getInstance().getExtensions().remove(FrameworkConstants.Config.QNAME_EXT_STEP_HANDLER);
-        FrameworkUtils.getStepHandler();
-        verify(GraphBasedStepHandler.class, times(1));
+        StepHandler stepHandler = FrameworkUtils.getStepHandler();
+        assertNotNull(stepHandler);
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:270]: check class name

**Comment by Thumimku** [comment]
Remove Identity Framework Testutils

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp [hunk 4]
```diff
@@ -521,14 +522,16 @@
                         </label>
                     </div>
                 </div>
-
                 <div style="clear:both"></div>
                 <!-- sectionSub Div -->
                 <br/>
+                <%
+                    if(FrameworkUtils.isAdaptiveAuthenticationAvailable()) {
+                %>
```
Comment (senthalan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp:530]: how we are going to handle this in our console ?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 5]
```diff
@@ -214,7 +214,13 @@ protected void activate(ComponentContext ctxt) {
         dataHolder.setJsFunctionRegistry(new JsFunctionRegistryImpl());
         BundleContext bundleContext = ctxt.getBundleContext();
 
-        bundleContext.registerService(JsFunctionRegistry.class, dataHolder.getJsFunctionRegistry(), null);
+        if (FrameworkUtils.isAdaptiveAuthenticationAvailable()) {
+            log.info("is Adaptive Authentication Enable : True");
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java:218]: let's improve this log to more user friendly

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 6]
```diff
@@ -2417,6 +2417,23 @@ public static String getUserNameProvisioningUIUrl() {
...
+
+        boolean isAdaptiveAuthenticationAvailable = false;
+        try {
+            Class.forName("org.openjdk.nashorn.api.scripting.ScriptObjectMirror");
+            isAdaptiveAuthenticationAvailable = true;
+            return isAdaptiveAuthenticationAvailable;
+        } catch (ClassNotFoundException e) {
+            return isAdaptiveAuthenticationAvailable;
+        }
+    }
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java:2435]: tring to do Class.forName each time is expensive, let's do this during the internal component activate logic and store the result in Dataholder and expose the value through this util.

File: components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java [hunk 7]
```diff
@@ -35,6 +35,11 @@ public class CacheBackedProvisioningMgtDAO {
 
     private ProvisioningEntityCache provisioningEntityCache = null;
 
+    public CacheBackedProvisioningMgtDAO() {
+        this.provisioningMgtDAO = null;
+        this.provisioningEntityCache = null;
+    }
+
```
Comment (senthalan) [inline components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java:42]: what is the reason for this constructor with null initialization ?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 8]
```diff
@@ -181,6 +181,8 @@ public abstract class FrameworkConstants {
     public static final String H2 = "H2";
     public static final String ORACLE = "Oracle";
 
+    public static final String ENABLE_ADAPTIVE_AUTHENTICATION = "AdaptiveAuthentication.Enable";
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:184]: Seems like an unused constant?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp (hunk 4 above)
Comment (Thumimku) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp:530]: with this PR https://github.com/wso2/identity-apps/pull/3116

File: identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/pom.xml [hunk 9]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (piraveena) [inline identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/pom.xml:3]: change the year in the license txt

File: identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/src/main/resources/repository/conf/carbon.xml [hunk 10]
```diff
@@ -0,0 +1,684 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (piraveena) [inline identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/src/main/resources/repository/conf/carbon.xml:3]: change year

File: identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/src/main/resources/repository/conf/identity/identity.xml [hunk 11]
```diff
@@ -0,0 +1,744 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (piraveena) [inline identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/src/main/resources/repository/conf/identity/identity.xml:3]: change year

Comment (piraveena) [reply identity-framework-test-utils/org.wso2.carbon.identity.framework.testutil/src/main/resources/repository/conf/identity/identity.xml]: Fix in other places as well

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java (hunk 6 above)
Comment (Thumimku) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Fixed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java (hunk 5 above)
Comment (Thumimku) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java]: +1

File: components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java (hunk 7 above)
Comment (Thumimku) [reply components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java]: removed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 12]
```diff
@@ -214,7 +214,16 @@ protected void activate(ComponentContext ctxt) {
         dataHolder.setJsFunctionRegistry(new JsFunctionRegistryImpl());
         BundleContext bundleContext = ctxt.getBundleContext();
 
-        bundleContext.registerService(JsFunctionRegistry.class, dataHolder.getJsFunctionRegistry(), null);
+        if (FrameworkUtils.checkAdaptiveAuthenticationAvailable()) {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java]: ack

**Review by senthalan — APPROVED** [review]

**Review by piraveena — APPROVED** [review]

**Pull request merged**